controller/api/public/grpc_server.go

```diff
@@ -94,7 +108,6 @@ func (s *grpcServer) ListPods(ctx context.Context, req *pb.ListPodsRequest) (*pb
 	}
 
 	var pods []*k8sV1.Pod
-	namespace := req.GetNamespace()
 	if namespace != "" {
 		pods, err = s.k8sAPI.Pod().Lister().Pods(namespace).List(labels.Everything())
```

Comment:
Should we also populate the label selector used here based on the "selector" object that's sent with the request?

For example, if I issue the following `ListPods` request, I'd expect to only get back only those pods with the `app: emoji-svc` label:

```
	req := &pb.ListPodsRequest{
		Selector: &pb.ResourceSelection{
			LabelSelector: "app=emoji-svc",
		},
	}
```

You can use `labels.Parse` to parse the label selector string. Something like:

```diff
diff --git a/controller/api/public/grpc_server.go b/controller/api/public/grpc_server.go
index 1740946d..522d2f21 100644
--- a/controller/api/public/grpc_server.go
+++ b/controller/api/public/grpc_server.go
@@ -107,11 +107,20 @@ func (s *grpcServer) ListPods(ctx context.Context, req *pb.ListPodsRequest) (*pb
 		}
 	}
 
+	labelSelector := labels.Everything()
+	if s := req.GetSelector().GetLabelSelector(); s != "" {
+		var err error
+		labelSelector, err = labels.Parse(s)
+		if err != nil {
+			return nil, fmt.Errorf("invalid label selector \"%s\": %s", s, err)
+		}
+	}
+
 	var pods []*k8sV1.Pod
 	if namespace != "" {
-		pods, err = s.k8sAPI.Pod().Lister().Pods(namespace).List(labels.Everything())
+		pods, err = s.k8sAPI.Pod().Lister().Pods(namespace).List(labelSelector)
 	} else {
-		pods, err = s.k8sAPI.Pod().Lister().List(labels.Everything())
+		pods, err = s.k8sAPI.Pod().Lister().List(labelSelector)
 	}
 
 	if err != nil {
```